### PR TITLE
Deep stacks for Pyret artifact

### PR DESCRIPTION
**Pull request by rachitnigam**

### CODE REVIEW

File: stopify-continuations/src/runtime/abstractRuntime.ts [hunk 1]
```diff
@@ -142,16 +157,94 @@ export abstract class Runtime {
...
-  abstract handleNew(constr: any, ...args: any[]): any;
-  abstract abstractRun(body: () => any): RunResult;
+/**
+ * Represents a runtime that does not support heap bounded stacks.
+ *
+ * Abstractly, the stack is restored top-down, i.e. from the most recent
+ * frame to the oldest frame. The topmost frame is popped off, run, and its
+ * value is threaded into the next frame.
+ */
+export abstract class DeepRuntime extends Runtime {
```
Comment (arjunguha) [inline stopify-continuations/src/runtime/abstractRuntime.ts]: Shallow and deep runtime both have the same comment.

File: stopify-continuations/src/runtime/abstractRuntime.ts [hunk 2]
```diff
@@ -142,16 +157,94 @@ export abstract class Runtime {
...
+            return this.stack[this.stack.length - 1].f()
+          }
+        }
+        else {
+          assert(this.mode, 'execution completed in restore mode')
+          return
+        }
+      }
+			else if (result.type === 'capture') {
+        const newStack = result.stack;
```
Comment (arjunguha) [inline stopify-continuations/src/runtime/abstractRuntime.ts]: What is up with this indentation?

**Comment by rachitnigam** [comment]
Don’t review this.

**Comment by rachitnigam** [comment]
No more commits to this branch. The pyret artifact is using it.

**Pull request closed without merging**